tests/unit/jobserver/views/test_workspaces.py

```diff
@@ -167,7 +167,7 @@ def test_workspacebackendfiles_with_permission(rf, project_membership, role_fact
     project_membership(
         project=workspace.project,
         user=user,
-        roles=[role_factory(permissions=[permissions.unreleased_outputs_view])],
+        roles=[role_factory(permission=permissions.unreleased_outputs_view)],
```

Comment:
The number of lines you've had to update in this file for such a simple change suggests there's something that can be improved here, either with the UserFactory / project_membership fixture or the tests or workspaces.py (is it doing too much?). I don't think you should do anything more now, but there's maybe a hint towards a future refactoring that's possible here. Same with releases and job_requests above, too.

Reply:
You're right, but that something isn't immediately clear (to me, at least). When working through the tests, I was struck by how many very similar tests we have. Possibly we could move the permissions check into a decorator, and then just test the decorator?